services/src/main/java/org/jboss/windup/web/services/rest/MigrationProjectEndpointImpl.java

```diff
@@ -96,8 +96,14 @@ public void deleteProject(MigrationProject migrationProject)
     }
 
     @Override
+    public void deleteOldProvisionalProjects()
+    {
+        this.migrationProjectService.deleteOldProvisionalProjects(0);
+    }
+
+    @Override
     public Long getProjectIdByName(String title){
-        String jql = "SELECT id FROM MigrationProject p WHERE LOWER(p.title) = LOWER(:title)";
+        String jql = "SELECT id FROM MigrationProject p WHERE LOWER(p.title) = LOWER(:title) AND p.provisional = FALSE";
```

Comment:
Hmm, duplicate handling is complicated isn't it? Maye the unique constraint can be extended to check both the title and the provisional field?

The validator could then ignore ones where "provisional = true", but it wouldn't be possible to save two of them as non-provisional.